component/src/main/java/org/wso2/extension/siddhi/store/redis/RedisTable.java

```diff
@@ -89,29 +97,55 @@
                 @Parameter(name = "password",
                         description = "password to connect to redis server",
                         type = {DataType.STRING}, optional = true,
-                        defaultValue = "will try to connect without authentication")
+                        defaultValue = "will try to connect without authentication"),
+                @Parameter(name = "cluster.mode",
+                        description = "This will decide the redis mode. if this is false, client will connect to a " +
+                                "single redis node.",
+                        type = {DataType.BOOL},
+                        defaultValue = "false"),
+                @Parameter(name = "nodes",
+                        description = "host and the port of Master nodes of the cluster. If this is not provided in " +
+                                "when the redis deployment is a clustered deployment",
+                        type = {DataType.STRING}, optional = true,
+                        defaultValue = "null"),
         },
         examples = {
                 @Example(
-                        syntax = "@store(type='redis',host='localhost',port=6379, password='root',table" +
-                                ".name='fooTable')" +
+                        syntax = "@store(type='redis',host='localhost',port=6379,password='root'," +
+                                "table.name='fooTable',cluster.mode=false)" +
                                 "define table fooTable(time long, date String)",
-                        description = "above collection will create a redis table with the name fooTable"
+                        description = "Above example will create a redis table with the name fooTable and work on a" +
+                                "single redis node."
+                ),
+                @Example(
+                        syntax = "@Store(type='redis', table.name='SweetProductionTable', " +
+                                "nodes='localhost:30001,localhost:30002,localhost:30003', cluster.mode='true')\n" +
+                                "@primaryKey('symbol')\n" +
```

Comment:
Why not use nodes for both clusters and single node?

Reply:
if we changed the host into nodes It won't backward compatible..

Reply:
Done the requested fix. nodes can be used in cluster mode as well and host, port also there in implementation.